File: ThreeDSecure/src/main/java/com/braintreepayments/api/interfaces/ThreeDSecurePrepareLookupCallback.java
package com.braintreepayments.api.interfaces;

import com.braintreepayments.api.models.ThreeDSecureRequest;

/**
 * Interface for ThreeDSecure prepareLookup callbacks.
 */
public interface ThreeDSecurePrepareLookupCallback {
    /**
     * Called when the 3DS preparation is completed.
     *
     * @param request the {@link ThreeDSecureRequest} with a nonce.
     * @param clientData JSON string of client data to be sent to server for lookup.
     */
    void onResult(ThreeDSecureRequest request, String clientData, Exception error);
}

File: ThreeDSecure/src/main/java/com/braintreepayments/api/threedsecure/ThreeDSecurePrepareLookupCallback.kt
package com.braintreepayments.api.threedsecure

/**
 * Callback for receiving result of [ThreeDSecureClient.prepareLookup].
 */
fun interface ThreeDSecurePrepareLookupCallback {

    /**
     * @param prepareLookupResult [ThreeDSecurePrepareLookupResult]
     */
    fun onPrepareLookupResult(prepareLookupResult: ThreeDSecurePrepareLookupResult?)
}
